src/ensembl/src/content/app/entity-viewer/gene-view/components/protein-domain-image/ProteinDomainImage.tsx

```diff
@@ -117,8 +118,11 @@ export const ProteinDomainImageWithData = (
   const proteinDomainsResources = getDomainsByResourceGroups(
     protein.protein_domains_resources
   );
+
+  const refProteinLength = refCDSLength / 3;
+
   const scale = scaleLinear()
-    .domain([1, protein.length])
+    .domain([0, refProteinLength])
```

Comment:
I believe this is wrong? If I understand what's going on correctly:
- your domain is the total number of amino acids in your protein; this should remain the same
- what's changing is the output range into which you have to fit your protein.

Here's a diagram:

![image](https://user-images.githubusercontent.com/6834224/86122471-d0111b00-bacf-11ea-83ad-6920f1ea3c93.png)

Reply:
In fact, the width that is to be used as a range in the protein domain component can probably be passed from the exons image component like so:

![image](https://user-images.githubusercontent.com/6834224/86122989-9f7db100-bad0-11ea-8f64-c932e1fabe23.png)

Reply:
On the other hand, after a second thought, your implementation should work too. It is probably easier than what I am suggesting as well. It may be a bit less intuitive in terms of what is being calculated here.